src/components/user-avatar.component.jsx

```diff
@@ -1,11 +1,26 @@
 import * as React from 'react';
 
+import { Link } from 'react-router';
 import Avatar from 'material-ui/Avatar';
 import SocialPerson from 'material-ui/svg-icons/social/person';
+import IconMenu from 'material-ui/IconMenu';
+import MenuItem from 'material-ui/MenuItem';
+import IconButton from 'material-ui/IconButton';
 
-const UserAvatar = ({ user, onTouchTap, style }) => (
-    user.vkPhotoUrl ? <Avatar src={user.vkPhotoUrl} style={style} onTouchTap={onTouchTap} />
-        : <Avatar icon={<SocialPerson />} style={style} onTouchTap={onTouchTap} />
+const UserAvatar = ({ user, style }) => (
+      <IconMenu
+          style={{marginRight: '2px'}}
+          iconButtonElement={
+              <IconButton style={{padding: '0px'}}>{
+                  user.vkPhotoUrl ? <Avatar src={user.vkPhotoUrl} style={style} /> :
+                  <Avatar icon={<SocialPerson />} style={style} />}
+              </IconButton>}
+          anchorOrigin={{horizontal: 'left', vertical: 'bottom'}}
+          targetOrigin={{horizontal: 'middle', vertical: 'top'}}
+          >
+          <MenuItem primaryText='Settings' containerElement={<Link to='/settings' />}/>
+          <MenuItem primaryText='Sign out' href='/logout'/>
```

Comment:
React router throws an error saying no route exists, which makes sense since we have no handler for it.[ This basically explains it. ](https://github.com/ReactTraining/react-router/issues/1147#issuecomment-113180174)